### ISSUE TITLE
Docker Emulator does not support hierachical keys

### ISSUE THREAD

**Issue by pstueck** [Azure-Samples/cosmicworks#26]
> Please provide us with the following information:
> ---------------------------------------------------------------

### This issue is for a: (mark with an `x`)
```
- [ ] bug report -> please search issues before submitting
- [ ] feature request
- [ ] documentation issue or request
- [x] regression (a behavior that used to work and stopped in a new release)
```
> It must have worked with `1.0.7` (this is the version that is used in some [DP-420 course labs](https://learn.microsoft.com/en-us/credentials/certifications/exams/dp-420)).

### Minimal steps to reproduce
> start docker CosmosDB emulator for Linux
install latest cosmicworks via `dotnet tool`
run `cosmicworks --emulator`
switch to emulator (e.g. `https://localhost:8081/_explorer/index.html`)
navigate to `Items` of one of the two containers
try to view an item
doesn’t work and shows problem information
`Failed to read item 00000000-0000-0000-0000-000000002001:`
`Partition key provided either doesn't correspond to definition in the collection or doesn't match partition key field values specified in the document.`
only `/company` is shown as partition key in the items list
trying to add a new item tells me the partition key does not fit (tried  with `/company` only, as well as with all three)

### Any log messages given by the failure
> none besides the (colorful) progress information

### Expected/desired behavior
> items in emulator are viewable
new items can be added
existing items can be altered

### OS and Version?
> Ubuntu 22.04 LTS
dotnet 7.0.115
cosmicworks 2.0.3
Docker 24.0.5
CosmosDB Linux docker image from `mcr.microsoft.com/cosmosdb/linux/azure-cosmos-emulator:latest`
Digest: sha256:85c7b1771de120d3923c077606383e52b3763a7fe740c9f13bb2a95b5cb841e9

### Mention any other details that might be useful
> With some GitHub/code research I learned that the samples use hierarchical keys (`/company,/department,/territory`), which the emulator does not support (at least I tried to manually create the containers and could not see an option for that).

> ---------------------------------------------------------------
> Thanks! We'll be in touch soon.

**Comment by seesharprun** [comment]
Thanks for reporting this!

I'll look around to see who can help with the emulator. In the interim, would a `--disable-hiearchical-partition-keys` command help resolve your issue?

**Comment by seesharprun** [comment]
I also just checked. DP-420 uses version 1.x of cosmicworks which has a separate functionality set than the latest version.

**Comment by pstueck** [comment]
> Thanks for reporting this!
> 
> I'll look around to see who can help with the emulator. In the interim, would a `--disable-hiearchical-partition-keys` command help resolve your issue?

Maybe I used that parameter the wrong way … yet … looks like cosmicworks 2.0.3 is not aware of that parameter, as neither invocation – with or without prefabricated containers – did work.

This is how I tried to generate the samples …
`cosmicworks --emulator --number-of-products 10 --number-of-employees 10 --disable-hiearchical-partition-keys`  
1. call was issued against an empty emulator (Linux Docker variant) … reading the items showed the “failed to read” error from above
1. call was issued against the database created in 1., with removed and recreated containers (e.g. `employees` with partition key `/company`)  
resulted in partition key mismatch error `Error: The requested partition key path '/company,/department,/territory' does not match existing Container 'employees' with partition key path '/company' 
(Parameter 'PartitionKey')`

Also did a `cosmicworks` call with `-c <connection string>`, in case `--emulator` ignores any parameters → no dice either.

**Comment by seesharprun** [comment]
I completely get it. The `--disable-hiearchical-partition-keys` doesn't exist yet, I was just asking if you thought that would be worth the effort.

There's some other improvements I want to make to this tool (namely RBAC support) and that could be included in a 2.1 variant.

**Comment by seesharprun** [comment]
This is now implemented
